app/models/financialDetails/WhatYouOweChargesList.scala

```diff
@@ -34,10 +33,7 @@ case class WhatYouOweChargesList(balanceDetails: BalanceDetails, overduePaymentL
       && outstandingChargesModel.get.bcdChargeType.get.chargeAmount > 0) true
     else false
 
-  def codedOutDocumentDetailNonEmpty: Boolean =
-    if (codedOutDocumentDetail.isDefined && codedOutDocumentDetail.get.amountCodedOut.isDefined
-      && codedOutDocumentDetail.get.amountCodedOut.get > 0) true
-    else false
+  def codedOutDocumentDetailNonEmpty: Boolean = codedOutDocumentDetail.fold(false)(_ => true)
```

Comment:
might be clearer like this:
if(codedOutDocumentDetail.isDefined) true else false